Mathlib/Tactic/RCases.lean

```diff
@@ -313,7 +313,7 @@ partial def rcasesCore (g : MVarId) (fs : FVarSubst) (clears : Array FVarId) (e
     unless ← isDefEq etype expected do
       Term.throwTypeMismatchError "rcases: scrutinee" expected etype e
     let g ← replaceLocalDeclDefEq g e.fvarId! expected
-    cont g fs clears a
+    rcasesCore g fs clears e.fvarId! a pat cont
```

Comment:
This is actually a code change, a bug that was caught by the unused variables linter. I added a test to check this.